front/src/common/UserSettings.tsx

```diff
@@ -76,6 +85,22 @@ export default function UserSettings() {
             <p className="ml-3">TrainSchedule V2</p>
           </div>
         </div>
+        <div className="col-lg-6">
+          <div className="d-flex align-items-center mt-3">
+            <SwitchSNCF
+              id="stdcm-version-switch"
+              type={SWITCH_TYPES.switch}
+              name="stdcm-version-switch"
+              onChange={() => {
+                dispatch(switchStdcmV2Activated());
+                dispatch(updateScenarioID(undefined));
+                dispatch(updateTimetableID(undefined));
```

Comment:
I think you can also reset the infraId
```suggestion
                dispatch(updateTimetableID(undefined));
                dispatch(updateInfraID(undefined));
```